e_ductus/e_ductus/fields.py

```diff
@@ -0,0 +1,29 @@
+from django.db import models
+from django.core.exceptions import ObjectDoesNotExist
+
+
+class OrderField(models.PositiveIntegerField):
+    def __init__(self, for_fields=None, *args, **kwargs):
+        self.for_fields = for_fields
+        super().__init__(*args, **kwargs)
+
+    def pre_save(self, model_instance, add):
+        if getattr(model_instance, self.attname) is None:
+            # no current value
+            try:
+                qs = self.model.objects.all()
+                if self.for_fields:
+                    # filter by objects with the same field values
+                    # for the fields in "for_fields"
+                    query = {field: getattr(model_instance, field) \
+                             for field in self.for_fields}
+                    qs = qs.filter(**query)
+                # get the order of the last item
+                last_item = qs.latest(self.attname)
+                value = last_item.order + 1
```

Comment:
It looks, that here might be issues with race condition.

https://docs.djangoproject.com/en/3.2/ref/models/expressions/#avoiding-race-conditions-using-f